src/bootstrap/src/core/sanity.rs

```diff
@@ -177,6 +178,34 @@ than building it.
             continue;
         }
 
+        // Check if there exists a built-in target in the list of supported targets.
+        let mut has_target = false;
+        let target_str = target.to_string();
+
+        let supported_target_list =
+            output(Command::new(&build.config.initial_rustc).args(["--print", "target-list"]));
```

Comment:
As I said on the associated issue, asking the bootstrap compiler is not correct when bootstrap doesn't know about a target yet, but the in-tree rustc does. You have to check if the target is listed in `compiler/rustc_target/src/spec/mod.rs`.